pkg/kubecost/asset.go

```diff
@@ -2996,9 +3034,10 @@ func (asr *AssetSetRange) InsertRange(that *AssetSetRange) error {
 
 // IsEmpty returns false if AssetSetRange contains a single AssetSet that is not empty
 func (asr *AssetSetRange) IsEmpty() bool {
-	if asr == nil || asr.Length() == 0 {
```

Comment:
asr.Length() call acquires and relinquishes RLock. Also, not needed, as a range loop will not iterate over 0 length collection (or nil).